enhancements/monitoring/alerting-consistency.md

```diff
@@ -0,0 +1,220 @@
+of the cluster or running applications.
+
+You might be thinking to yourself: "But my component is **super** important, and
+if it's not working, the user can't do X, which user definitely will want."  All
+that is probably true, but that doesn't make your alert "critical" it just makes
+it worthy of an alert.
+
+### Warning Alerts
+
+TL/DR: fix this soon, or some things won't work and upgrades will probably be
+blocked.
+
+If your alert does not meet the criteria in "Critical Alerts" above, it belongs
+to the warning level or lower.
+
+Use warning level alerts for reporting conditions that may lead to inability to
+deliver individual features of the cluster, but not service for the cluster as a
+whole. Most alerts are likely to be warnings. Configure warning level alerts so
+that they do not fire until components have sufficient time to try to recover
+from the interruption automatically. Expect users to be notified of a warning,
+but for them not to respond with corrective action immediately.
+
+Timeline:  ~60 minutes
+
+60 minutes?  That seems high!  That's because it is high.  We want to reduce
+the noise.  We've done a lot to make clusters and operators auto-heal
+themselves.  The whole idea is that if a condition has persisted for more than
+60 minutes, it's unlikely to be resolved without intervention any time later.
+
+#### Example 1: All machine-config-daemons are crashlooping.
+
+That seems significant
+and it probably is, but nobody has to get out of bed to fix this **right now**.
+But, what if a machine dies, I won't be able to get a replacement?  Yeah, that
+is probably true, but how often does that happen?  Also, that's an entirely
+separate set of concerns.
+
+#### Example 2:  I have an operator that needs at least 3 of something to work.
+* With 2/3 replicas, warning alert after 60M.
+* With 1/3 replicas, warning alert after 10M.
+* With 0/3 replicas, warning alert after 5M.
+
+Q: How long should we wait until the above conditions rises to a 'critical'
+alert?
+
+A: It should never rise to the level of a critical alert, it's not critical.  If
+it was, this section would not apply.
+
+#### Example 3:  I have an operator that only needs 1 replica to function
+
+Well, if the cluster can upgrade with only 1 replica, and the the service is
+not unavailable despite other replicas being unavailable, this can probably
+be just an info-level alert.  We should have a "minor" alert, TBD.  But for now,
+"info" is the closest thing.
+
+#### Q: What if a particular condition will block upgrades?
+
+A: It's a warning level alert.
+
+
+### Alert Ownership
+
+Previously, the bulk of our alerting was handled directly by the monitoring
+team.  Initially, this gave use some indication into platform status without
+much effort by each component team.  As we mature our alerting system, we should
+ensure all teams take ownership of alerts for their individual components.
+
+Going forward, teams will be expected to own alerting rules within their own
```

Comment:
We should enumerate the ownership responsibilities, including

1. first in line to receive the bug for an alert firing
2. responsible for describing, "what does it mean" and "what does it do"
3. responsible for choosing the level of alert
4. responsible for deciding if the alert even matters

I think we could get some improvement by categorizing alerts by owner so we can push for better refinement, but that could be a later step.

Reply:
:+1: totally agree with what David write, but assuming we have clear guidelines about critical/warning/info alert levels described above in this document.

Reply:
As I said earlier, I don't think writing this in this PR constitutes active acceptance and acknowledgement by the stakeholders.  Want to avoid surprising folks with new responsibilities they had no heads-up to.